packages/frint-props/README.md

```diff
@@ -0,0 +1,384 @@
+
+```js
+import { compose, withDefaults, withState, shouldUpdate } from 'frint-props';
+
+const props$ = compose(
+  withDefaults({ counter: 0 }),
+  withState('counter', 'setCounter', 0),
+  shouldUpdate((prevProps, nextProps) => {
+    return prevProps.counter !== nextProps.counter;
+  })
+)();
+```
+
+The implementation of `shouldUpdate` above tells our `props$` Observable to emit new values only if the `counter` value has changed. Otherwise nothing new is emitted.
+
+## Write your own functions
+
+The API of creating your own functions is pretty simple.
+
+### Basic function to add props
+
+A basic function that is concerned about adding new props can be written like this:
+
+```js
+function withFoo() {
+  return function () {
+    return {
+      foo: 'foo value here',
+    };
+  };
+}
+```
+
+You can also return an Observable instead of a plain object:
+
+```js
+import { of } from 'rxjs/observable/of';
+
+function withFoo() {
+  return function () {
+    return of({
+      foo: 'foo value here',
+    });
+  };
+}
+```
+
+### Accessing additional arguments
+
+You would notice that instead of returning the Object/Observable directly from our function, we return another function which takes care of returning the final result.
+
+This allows us to access additional arguments when composing props.
+
+Imagine if we want to make the FrintJS `app` instance available to our functions:
+
+```js
+const app = new App();
+
+const props$ = compose(
+  withFoo()
+)(app);
+```
+
+The argument `app` can now accessed inside our function like this:
+
+```js
+withFoo() {
+  return function (app) {
+    return {
+      foo: 'foo value here',
+    };
+  };
+}
+```
+
+The way some of our functions are designed in this repository, the returned functions expect to receive the same arguments as our `observe` higher-order component receives in `frint-react`, which are: `app` and `parentProps$`.
+
+### Operator functions
+
+Besides just adding new props, functions can also take care of processing the stream further just like how RxJS operators work.
+
+We can create a function, that will check if there is any `foo` prop, and then capitalize it:
+
+```js
+import { map } from 'rxjs/operators/map';
+
+function capitalizeFoo() {
+  return function () {
+    return map((props) => ({
+      ...props,
+      foo: props.foo
+        ? props.foo.toUpperCase()
+        : undefined,
+    }));
+  };
+}
+```
+
+Can be composed together as follows:
+
+```js
+import { compose } from 'frint-props';
+
+const props$ = compose(
+  withFoo(),
+  capitalizeFoo()
+);
+```
+
+The `props$` Observable will now emit `{ foo: 'FOO VALUE HERE' }`.
+
+---
+
+# API
+
+All the functions return a function, when called, returns an Observable of props.
+
+<!-- -->
+
+## withDefaults
+
+> withDefaults(defaultProps)
+
+### Arguments
+
+1. `defaultProps`: Default props to start the stream with
+
+<!-- -->
+
+## withState
```

Comment:
Is this part duplicated from the withState defined at line 271? Also the example of this part belongs to the withDefaults section above this one

Reply:
yep! now removed.